### PR TITLE
Adding Regionalgam suite tool

### PR DESCRIPTION
**Pull request by Alanamosse**
This tool suite follows the work and tutorial created by Reto Schmucki.
Source : https://github.com/RetoSchmucki/regionalGAM/tree/v.1.5

7 tools : 7xml + 7Rscript
1 Rscript functions source file
1 shed file
1 macro file
11 test-data/testfiles

### CODE REVIEW

File: tools/regionalgam/.shed.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+long_description: Implementation of a method for computing an abundance index with the regional GAM approach, an extension of the two-stages model introduced by Dennis et al. (2013). This index offers the best performance for a wide range of sampling frequency, providing greater robustness and unbiased estimates compared to the popular linear interpolation approach (Schmucki et al. 2015).
+
+homepage_url: https://github.com/RetoSchmucki/regionalGAM 
+name: regionalgam
+owner: ecology 
+remote_repository_url: https://github.com/galaxyecology/tools-ecology/tools/regionalgam
+type: unrestricted
+
+repositories:
+  regionalgam_ab-index:
```
Comment (bgruening) [inline tools/regionalgam/.shed.yml]: Is this also needed in this repo? I know it was needed in the other repo, but we should really avoid this. Think about it this way. Even if testing turns green and everything works perfectly, if this list is not up-to-date the TS installation will be broken and we have no way to test this atm :(

File: tools/regionalgam/ab-index.xml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+<tool id="regionalgam_ab-index" name="Abundance index" version="@VERSION@">
```
Comment (bgruening) [inline tools/regionalgam/ab-index.xml]: I would always use `_`

File: tools/regionalgam/flight-curve.xml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        <data format="tabular" name="output" from_work_dir="pheno" />
+    </outputs>
+    <tests>
+        <test>
+            <param name="count_file" value="gatekeeper_CM"/>
+            <param name="minvisit" value="3"/>
+            <param name="minoccur" value="2"/>
+            <output name="output" ftype="tabular">
+                <assert_contents>
+                    <has_line line="&quot;species&quot;&#009;&quot;year&quot;&#009;&quot;week&quot;&#009;&quot;DAYNO&quot;&#009;&quot;DAYNO_adj&quot;&#009;&quot;nm&quot;"/>
```
Comment (bgruening) [inline tools/regionalgam/flight-curve.xml]: Maybe you can also use CDATA here?

File: tools/regionalgam/ab-index.xml (hunk 2 above)
Comment (Alanamosse) [reply tools/regionalgam/ab-index.xml]: ok !

File: tools/regionalgam/.shed.yml (hunk 1 above)
Comment (Alanamosse) [reply tools/regionalgam/.shed.yml]: Ok I try a tool suite, Thank for your comments

**Comment by Alanamosse** [comment]
Thks @bgruening for your comments :)
 - I'm reworking the tests, so no problem ! it's a good occasion to reduce this file size
 - For the version I'm referring to the author source release : https://github.com/RetoSchmucki/regionalGAM/releases

**Comment by Alanamosse** [comment]
Hi @bgruening 
I need some help please 2 remaining tests failed :( 
That's because the corresponding tools generate png and those outputs files can be slightly different across 2 executions (random effects in the analyse). 
Do you know any way in `<test>`  section to just check whether the output file is a png without executing the `diff` command.
The `<assess_contents>` is useful for text files, I have no clue for binary thought.

**Comment by bgruening** [comment]
Hi you `sim_size` as compare method. That should work.

**Comment by bgruening** [comment]
> /tmp/tmpa6basus_res.png=4575b but /tmp/tmpW15mmIs_res.png=2632b - compare by size (delta=100) failed

This is really a large difference.

**Comment by Alanamosse** [comment]
I agree there is something wrong.. 
A `compare diff` works with planemo on my side but somehow the output png files are very different with travis tests.
Maybe I missed some conda dep

**Pull request closed without merging**